newrelic-agent/src/main/java/com/newrelic/agent/rpm/RPMConnectionServiceImpl.java

```diff
@@ -45,12 +44,13 @@ public class RPMConnectionServiceImpl extends AbstractService implements RPMConn
     public static final long APP_SERVER_PORT_TIMEOUT = 120L;
     // interval between connection attempts
     public static final long MIN_CONNECT_ATTEMPT_INTERVAL = 5L;
-    private final ScheduledExecutorService scheduledExecutor;
+    private static final int MAX_QUEUED_CONNECT_TASKS = 5000;
+    private final ScheduledThreadPoolExecutor scheduledExecutor;
```

Comment:
I would switch this back to `ScheduledExecutorService` so that wherever we use `scheduledExecutor` we are dealing with an interface rather than a specific implementation of that interface. In general, this will help prevent us from becoming too dependent on a specific implementation (with the one exception called out below) which will make it easier to change to a different implementation if we should need to do so at some point in the future. As unlikely as that may be it's a good practice to follow.